torch_xla/csrc/ops/expand_symint.cpp

```diff
@@ -0,0 +1,66 @@
+#include "torch_xla/csrc/ops/expand_symint.h"
+
+#include "absl/strings/str_join.h"
+#include "tensorflow/compiler/xla/client/lib/constants.h"
+#include "tensorflow/compiler/xla/xla_client/debug_macros.h"
+#include "torch/csrc/lazy/core/helpers.h"
+#include "torch/csrc/lazy/core/util.h"
+#include "torch_xla/csrc/data_ops.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+
+namespace torch_xla {
+namespace {
+
+xla::Shape NodeOutputShape(const torch::lazy::Value& input,
+                           const std::vector<int64_t> upper_bounds,
+                           const std::vector<bool> dynamic_dims) {
+  return xla::ShapeUtil::MakeShape(GetXlaShape(input).element_type(),
+                                   {upper_bounds}, {dynamic_dims});
+}
+
+std::vector<torch::lazy::Value> GetValues(
+    const torch::lazy::Value& input,
+    const std::vector<torch::lazy::Value>& dimensions) {
+  std::vector<torch::lazy::Value> values = dimensions;
+  values.insert(values.begin(), input);
+  return values;
+}
+
+}  // namespace
+
+ExpandSymInt::ExpandSymInt(const torch::lazy::Value& input,
+                           const SymIntElements& size_elements,
+                           const torch::lazy::Shape& shape)
+    : XlaNode(torch::lazy::OpKind(at::aten::expand),
+              GetValues(input, size_elements.GetValues()), {shape},
+              [&]() {
+                return NodeOutputShape(input, size_elements.GetUpperBounds(),
+                                       size_elements.GetDynamicDims());
+              },
+              /*num_outputs=*/1,
+              torch::lazy::MHash(size_elements.GetUpperBounds(),
+                                 size_elements.GetDynamicDims())),
+      upper_bounds_(std::move(size_elements.GetUpperBounds())),
+      dynamic_dims_(std::move(size_elements.GetDynamicDims())) {}
+
+XlaOpVector ExpandSymInt::Lower(LoweringContext* loctx) const {
+  xla::XlaOp input = loctx->GetOutputOp(operand(0));
+  std::vector<xla::XlaOp> size_ops;
+  for (int i = 1; i < operands().size(); i++) {
+    size_ops.push_back(loctx->GetOutputOp(operand(i)));
+  }
+  xla::XlaOp output =
+      SetDimensionSizes(BuildExpand(input, upper_bounds_), size_ops);
```

Comment:
I think you want to pass `dynamic_dims_` and use that to determine if you want to call `SetDimensionSizes`